azkaban-common/src/main/bash/util.sh

```diff
@@ -1,57 +1,55 @@
 #!/usr/bin/env bash
 # Common utils
-set -o nounset
+set -o nounset   # exit the script if you try to use an uninitialised variable
+set -o errexit   # exit the script if any statement returns a non-true return value
 
-
-# If process is still running then abort start up
-function abort_if_process_already_running {
-  installdir="$1"
-  pidfile="${installdir}/currentpid"
-  if [ -f ${pidfile} ]; then
-    pid=`cat ${pidfile}`
-    ps -p ${pid} 2>&1 1>nul && echo "Process already running [pid: ${pid}]. Aborting." && exit 1
-  fi
+#---
+# is_process_running: Checks if a process is running
+# args:               Process ID of running proccess
+# returns:            returns 0 if process is running, 1 if not found
+#---
+function is_process_running {
+  local  pid=$1
+  kill -0 $pid > /dev/null 2>&1 #exit code ($?) is 0 if pid is running, 1 if not running
+  local  status=$?              #because we are returning exit code, can use with if & no [ bracket 
+  return $status
 }
 
-# kill the process with retry
-# return 0 if kill succeeds or no process to kill,
-#        1 if kill fails
+#---
+# kill_process_with_retry: Reads pid file and attempts to kill the running process
+# args:                    None
+# returns:                 returns 0 if kill succeds or nothing to kill, 1 if kill fails
+#---
 function kill_process_with_retry {
-   pid="$1"
-   pname="$2"
-   maxattempt="$3"
-
-   if [[ -z $pid ]]; then
```

Comment:
Yes that's correct.  My thinking was the calling script would first call is_process_running before calling kill_process_with_retry, but failed to update azkaban-solo-shutdown.sh to demonstrate this behavior.   Now I'm questioning if that's appropriate behavior.

I'll update this PR by placing a guard inside the kill_process_with_retry function to prevent killing non-existent pids.